### PR TITLE
Fix up SELinux labeling

### PR DESCRIPTION
**Pull request by rhatdan**
SELinux label options processing fixes, should allow system tests to pass.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@baude PTAL

**Comment by baude** [comment]
LGTM

**Comment by baude** [comment]
@mheon bc this has specgen changes, you should ptal

File: pkg/specgen/container_validate.go [hunk 1]
```diff
@@ -68,18 +68,6 @@ func (s *SpecGenerator) Validate() error {
 	if len(s.CapAdd) > 0 && s.Privileged {
 		return exclusiveOptions("CapAdd", "privileged")
 	}
-	// selinuxprocesslabel and privileged are exclusive
```
Comment (mheon) [inline pkg/specgen/container_validate.go:71]: Why are we turning these off? Privileged flag disables all manual SELinux config, last I checked

File: pkg/specgen/specgen.go [hunk 2]
```diff
@@ -230,14 +230,6 @@ type ContainerSecurityConfig struct {
 	// If SELinux is enabled and this is not specified, a label will be
 	// automatically generated if not specified.
 	// Optional.
-	SelinuxProcessLabel string `json:"selinux_process_label,omitempty"`
```
Comment (mheon) [inline pkg/specgen/specgen.go:233]: Is it necessary to completely remove this?

File: pkg/specgen/container_validate.go (hunk 1 above)
Comment (rhatdan) [reply pkg/specgen/container_validate.go:71]: In the other code, we are setting the selinux options to "disabled" when --privileged mode is enabled.  I guess I could check over there.  This was causing an error to be thrown here.

File: pkg/specgen/specgen.go (hunk 2 above)
Comment (rhatdan) [reply pkg/specgen/specgen.go:233]: It is not used.

Comment (mheon) [reply pkg/specgen/specgen.go:233]: Why? The original intent of SpecGen was to make things as explicit as possible, removing parsing where possible - it seems like we're regressing on this here?

Comment (rhatdan) [reply pkg/specgen/specgen.go:233]: We don't know at the time that the specgen is created what the labels are going to be.  The labels are creates in container/storage which happens after this is generate.

Comment (mheon) [reply pkg/specgen/specgen.go:233]: These were originally intended for user-specified labels from `--security-opt label=` but it doesn't look like those compile cleanly into a complete process/mount label without outside intervention

**Comment by mheon** [comment]
Success failed, rerunning

LGTM otherwise

**Comment by rhatdan** [comment]
@mheon That is not the way `--security-opt label=` works.
It does
```
man podman create
...
       --security-opt=option

       Security Options

              · label=user:USER     : Set the label user for the container processes

              · label=role:ROLE     : Set the label role for the container processes

              · label=type:TYPE     : Set the label process type for the container processes

              · label=level:LEVEL   : Set the label level for the container processes

              · label=filetype:TYPE : Set the label file type for the container files

              · label=disable       : Turn off label separation for the container
```

**Comment by stevekuznetsov** [comment]
/hold
```
 Required status check \"success\" is cancelled
```

**Comment by rhatdan** [comment]
@stevekuznetsov What is going on?

**Comment by mheon** [comment]
@cevich Poke. Success task seems broken here. Rerunning not helping.

**Pull request merged**